src/communication.ts

```diff
@@ -12,55 +15,47 @@ export class Communication {
         });
     }
 
-    /**
-     * connect unregistered user to the website
-     * @param randomSeed the random seed the user need to connect on typically recieved in jwt
+    /* connects to the Tonomy Communication server, authenticates with it's DID
+     * subscribes to any messages that are sent by `sendMessage` by providing a callback function executed every time a message is received
+     * should send a read receipt when messages are received
+     * @returns true if successful
      */
-    connectTonomy(randomSeed: string) {
-        this.socketServer.emit('connectTonomy', { randomSeed });
-    }
+    login(authorization: Message): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+            this.socketServer.emit('login', { message: authorization.jwt }, (response: any) => {
+                if (response.err) {
+                    reject(response.err);
+                }
 
-    onClientConnected(func: (param?: any) => void) {
-        this.socketServer.on('connectTonomy', (param) => {
-            func(param);
+                resolve(response);
+            });
         });
     }
 
-    /**
-     * makes user login to websites
+    /* sends a message to another DID
+     * create a Message object from the message argument
+     * the message is used as the `vc` property of a VC signed by the User's key
      */
-    login(userName: string, client: string) {
-        this.socketServer.emit('loginTonomy', { client, userName });
-    }
+    sendMessage(message: Message): Promise<boolean> {
+        return new Promise((resolve, reject) => {
```

Comment:
probably should also only allow this if login has been successfully completed. New Task